Jenkinsfile

```diff
@@ -260,12 +260,28 @@ def k8sTest(Map args = [:]) {
   }
 }
 
+/**
+* This method is a wrapper to run the runCommand method and retry if there are 
+* environmental issues. Therefore it passes the arguments to the runCommand.
+* For further details regarding the arguments please refers to the runCommand method.
+*/
+def target(Map args = [:]) {
+  try {
+    runCommand(args)
+  } catch (err) {
+    if(fileExists('environmental-issue')) {
+      sleep 10
+      runCommand(args)
+    }
```

Comment:
The file is not deleted explicitly, it is true that runCommand deletes the folder but it is weird/hiden. Also, there is no stop condition, what happens if a tool is no longer available for installation? I think that will enter on an infinite loop

Reply:
- Actually the file is created in the top level agent, therefore this approach might cause issues for some other stages. It seems this approach might not be needed with files but with global variables. Good catch!
- No infinite loops at all, but two runs since it calls the `runCommand` rather than itself.